File: python/searchoption.py
################################################################################
#
# searchoption.py
#
# class SearchOption: encapsulates a (command-line) search option
#
################################################################################

class SearchOption:
    """a class to encapsulate a specific command line option"""
    def __init__(self, shortarg, longarg, desc, func):
        self.shortarg = shortarg
        self.longarg = longarg
        self.desc = desc
        self.func = func

    @property
    def sortarg(self):
        if self.shortarg:
            return self.shortarg.lower()
        else:
            return self.longarg.lower()

File: ruby/searchoption.rb
################################################################################
#
# searchoption.rb
#
# class SearchOption: encapsulates a (command-line) search option
#
################################################################################

class SearchOption

  attr_accessor :shortarg
  attr_accessor :longarg
  attr_accessor :desc
  attr_accessor :func

  def initialize(shortarg, longarg, desc, func)
    @shortarg = shortarg
    @longarg = longarg
    @desc = desc
    @func = func
  end

  def sortarg
    if not @shortarg.nil? and not @shortarg.empty?
        @shortarg.downcase
    else
      @longarg.downcase
    end
  end

end
